lib/static-files-router.js

```diff
@@ -0,0 +1,287 @@
+            response.addHeader("Content-type", "text/css");
+            response.addHeader("Cache-Control", "public,max-age=12960000");
+            response.addHeader("Last-Modified", String(lessFile.getLastModified()));
+            cachedFile.open('r');
+            print(cachedFile.getStream());
+            cachedFile.close();
+            return;
+        }
+
+        var less = getLessParser(configs);
+        // Adapted from https://github.com/less/less.js/blob/v1.7.5/lib/less/rhino.js#L149
+        var options = {
+            depends: false,
+            compress: false,
+            cleancss: false,
+            max_line_len: -1.0,
+            optimization: 1.0,
+            silent: false,
+            verbose: false,
+            lint: false,
+            paths: [],
+            color: true,
+            strictImports: false,
+            rootpath: "",
+            relativeUrls: false,
+            ieCompat: true,
+            strictMath: false,
+            strictUnits: false,
+            filename: lessFile.getPath()
+        };
+        var lessParser = less.Parser(options);
+
+        lessFile.open('r');
+        var lessCode = lessFile.readAll();
+        lessFile.close();
+        var callback = function (error, root) {
+            if (error) {
+                // something went wrong when processing the LESS file
+                var errorMsg = stringify(error);
+                log.warn("Failed to process '" + lessFile.getPath() + "' file due to " + errorMsg);
+                response.sendError(500, "Could not process LESS file. \n" + errorMsg);
+                return;
+            }
+
+            var result = root.toCSS(options);
+            cachedFile.open('w');
+            cachedFile.write(result);
+            cachedFile.close();
+
+            response.addHeader("Content-type", "text/css");
+            response.addHeader("Cache-Control", "public,max-age=12960000");
+            response.addHeader("Last-Modified", String(lessFile.getLastModified()));
+            print(result);
+        };
+        var globalVars = {"unit-class": splitFileName(unitFullName).extension}; // unit's short name
+        lessParser.parse(lessCode, callback, {globalVars: globalVars});
+    }
+
+    /**
+     * Process a HTTP request that requests a static file.
+     * @param request {Object} HTTP request to be processed
+     * @param response {Object} HTTP response to be served
+     * @param lookUpTable {Object}
+     * @param configs {Object} configurations
+     */
+    this.route = function (request, response, lookUpTable, configs) {
+        // URI = /{appName}/public/{unitFullName}/{resourceType}/{fileName}.{extension}
+        var uri = request.getRequestURI();
+        var parts = uri.split("/");
+        // parts= ["/", {appName}, "public", {unitFullName}, {resourceType}, {fileName}.{extension}]
+        if (parts.length != 6) {
+            // an invalid URI
+            log.warn("Request URI '" + uri + "' is invalid.");
```

Comment:
parts.length = 5 is a special case. should give a special error saying resources in public not allowed.